src/components/Graph/Graph.tsx

```diff
@@ -30,240 +45,298 @@ const isEdgeOnPatientPath = (pathwayResults: PathwayResults, edge: Edge): boolea
+      ? evaluatedPathway.pathwayResults.documentation
+      : [];
 
-  return (
-    <div
-      ref={graphElement}
-      id="graph-root"
-      className={styles.root}
-      style={{
-        height: interactive ? maxHeight + 150 : 'inherit',
-        width: maxWidth + 'px',
-        position: 'relative',
-        marginRight: '5px'
-      }}
-    >
-      {nodeCoordinates !== undefined
-        ? Object.keys(nodeCoordinates).map(key => {
-            const docResource = documentation.find((doc): doc is DocumentationResource => {
-              return typeof doc !== 'string' && doc.state === key;
-            });
-            const onClickHandler = interactive ? (): void => setExpanded(key) : undefined;
-            return (
-              <Node
-                key={key}
-                documentation={docResource}
-                ref={(node: HTMLDivElement): void => {
-                  nodeRefs.current[key] = node;
-                }}
-                pathwayState={pathway.states[key]}
-                isOnPatientPath={
-                  evaluatedPathway.pathwayResults
-                    ? evaluatedPathway.pathwayResults.path.includes(key) ||
-                      evaluatedPathway.pathwayResults.currentStates.includes(key)
-                    : false
-                }
-                isCurrentNode={
-                  evaluatedPathway.pathwayResults
-                    ? evaluatedPathway.pathwayResults.currentStates.includes(key)
-                    : false
-                }
-                xCoordinate={nodeCoordinates[key].x + parentWidth / 2}
-                yCoordinate={nodeCoordinates[key].y}
-                expanded={expanded[key]}
-                onClickHandler={onClickHandler}
-              />
-            );
-          })
-        : []}
-      <svg
-        xmlns="http://www.w3.org/2000/svg"
+    return (
+      <GraphRender
+        graphElement={graphElement}
+        interactive={interactive}
+        maxHeight={maxHeight}
+        nodeCoordinates={nodeCoordinates}
+        documentation={documentation}
+        edges={edges}
+        evaluatedPathway={evaluatedPathway}
+        nodeRefs={nodeRefs}
+        parentWidth={parentWidth}
+        maxWidth={maxWidth}
+        pathway={pathway}
+        expanded={expanded}
+        setExpanded={setExpanded}
+      />
+    );
+  }
+);
+
+interface GraphRenderProps {
+  graphElement: RefObject<HTMLDivElement>;
+  interactive: boolean;
+  maxHeight: number;
+  nodeCoordinates: NodeCoordinates;
+  documentation: (string | Documentation)[];
+  edges: Edges;
+  evaluatedPathway: EvaluatedPathway;
+  nodeRefs: React.MutableRefObject<{
+    [key: string]: HTMLDivElement;
+  }>;
+  parentWidth: number;
+  maxWidth: number;
+  pathway: Pathway;
+  expanded: {
+    [key: string]: boolean | undefined;
+  };
+  setExpanded: (key: string, expand?: boolean | undefined) => void;
+}
+
+const GraphRender: FC<GraphRenderProps> = memo(
```

Comment:
You probably hate me by now, but I think the suffix "Memo" would more accurately describe the component.  i.e. `GraphMemo`